## Task 2
### In this task you will work with data collected from our wheeled robot , check How_data_collected_for_Task_2.ipynb for more info.
### You have 7 txt files (0.txt to 6.txt) each represent a step
### build a map of the robot at each step independently using the code provided in Task 1 ( taken from http://ais.informatik.uni-freiburg.de/teaching/ss20/robotics/)

In [83]:
import numpy as np
import matplotlib.pyplot as plt
import math
pi = math.pi
#arr[]=np.array()

In [84]:
for i in range(7):### loop opens each file and prints number of measurments
    
    with open(str(i)+'.txt') as f:
        line_1 = f.readline()
        line_as_list=line_1.split('\t')
        arr = np.array(line_as_list)
        print(len(arr))
    f.close()
##### print  

100
73
107
66
122
99
56


In [85]:

## oppen one file and review data in array arr
with open(str(i)+'.txt') as f:
    line_1 = f.readline()
    line_as_list=line_1.split('\t')
    arr = np.array(line_as_list)
    print(len(arr))
    print(arr[0])# 
f.close()


56
(15, 280.296875, 1203.0)


###  What does (15, 280.296875, 1203.0) mean (check 
#### First value (mostly 15 ) is  quality : int which is  Reflected laser pulse strength

#### Second value is  angle : float

#### The measurment heading angle in degree unit [0, 360)

#### Third value is  distance : float

#### Measured object distance related to the sensor’s rotation center. In millimeter unit. Set to 0 when measurment is invalid.

In [ ]:
## Here comes your code 
## plot each 

In [31]:
text_file = open("0.txt", "r")
lines = text_file.readlines()
print (lines)
print (len(lines))
text_file.close()

['(15, 167.8125, 1295.25)\t(15, 89.484375, 489.75)\t(15, 91.625, 513.5)\t(15, 93.859375, 542.5)\t(15, 96.265625, 575.25)\t(15, 98.34375, 613.5)\t(15, 100.671875, 655.75)\t(15, 102.78125, 709.75)\t(15, 104.984375, 771.5)\t(15, 111.734375, 1048.25)\t(15, 114.109375, 1024.25)\t(15, 116.578125, 1008.5)\t(15, 119.0, 983.25)\t(15, 123.6875, 1056.25)\t(15, 125.984375, 1118.25)\t(15, 128.328125, 1077.5)\t(15, 130.84375, 990.25)\t(15, 133.1875, 972.25)\t(15, 135.5625, 980.0)\t(15, 138.0, 1006.5)\t(15, 140.25, 1071.25)\t(15, 142.578125, 1138.75)\t(15, 144.875, 1162.5)\t(15, 147.46875, 1020.5)\t(15, 149.578125, 1190.5)\t(15, 151.984375, 1209.5)\t(15, 154.3125, 1229.5)\t(15, 156.890625, 1253.75)\t(15, 159.1875, 1283.0)\t(15, 161.609375, 1319.0)\t(15, 163.96875, 1360.0)\t(15, 166.375, 1339.0)\t(15, 168.796875, 1266.75)\t(15, 171.25, 1202.25)\t(15, 173.71875, 1151.75)\t(15, 176.21875, 1039.5)\t(15, 178.390625, 1244.0)\t(15, 180.46875, 1306.5)\t(15, 182.84375, 1385.25)\t(15, 185.171875, 1463.5)\t(15,

In [36]:
lines

['(15, 167.8125, 1295.25)\t(15, 89.484375, 489.75)\t(15, 91.625, 513.5)\t(15, 93.859375, 542.5)\t(15, 96.265625, 575.25)\t(15, 98.34375, 613.5)\t(15, 100.671875, 655.75)\t(15, 102.78125, 709.75)\t(15, 104.984375, 771.5)\t(15, 111.734375, 1048.25)\t(15, 114.109375, 1024.25)\t(15, 116.578125, 1008.5)\t(15, 119.0, 983.25)\t(15, 123.6875, 1056.25)\t(15, 125.984375, 1118.25)\t(15, 128.328125, 1077.5)\t(15, 130.84375, 990.25)\t(15, 133.1875, 972.25)\t(15, 135.5625, 980.0)\t(15, 138.0, 1006.5)\t(15, 140.25, 1071.25)\t(15, 142.578125, 1138.75)\t(15, 144.875, 1162.5)\t(15, 147.46875, 1020.5)\t(15, 149.578125, 1190.5)\t(15, 151.984375, 1209.5)\t(15, 154.3125, 1229.5)\t(15, 156.890625, 1253.75)\t(15, 159.1875, 1283.0)\t(15, 161.609375, 1319.0)\t(15, 163.96875, 1360.0)\t(15, 166.375, 1339.0)\t(15, 168.796875, 1266.75)\t(15, 171.25, 1202.25)\t(15, 173.71875, 1151.75)\t(15, 176.21875, 1039.5)\t(15, 178.390625, 1244.0)\t(15, 180.46875, 1306.5)\t(15, 182.84375, 1385.25)\t(15, 185.171875, 1463.5)\t(15,

In [1]:
angle = np.linspace(-pi/2, pi/2, np.shape(scan)[0], endpoint='true')

In [1]:
x = scan * np.cos(angle); 
y = scan * np.sin(angle);
plt.plot(x, y, '.k', markersize=3)
# Set the same scale on both axes
plt.gca().set_aspect('equal')
plt.savefig('scan1.pdf')

ValueError: could not convert string to float: '(15,'